File: Blaze/Objects/BlazeInputTile.m
//
//  BlazeInputTile.m
//  Blaze
//
//  Created by Bob de Graaf on 07-10-15.
//  Copyright © 2015 GraafICT. All rights reserved.
//

#import "BlazeInputTile.h"

@implementation BlazeInputTile

-(id)initWithID:(int)ID text:(NSString *)text tintColor:(UIColor *)tintColor baseColor:(UIColor *)baseColor imageName:(NSString *)imageName
{
    return [self initWithID:ID name:nil text:text tintColor:tintColor baseColor:baseColor imageName:imageName];
}

-(id)initWithID:(int)ID name:(NSString *)name text:(NSString *)text tintColor:(UIColor *)tintColor baseColor:(UIColor *)baseColor imageName:(NSString *)imageName
{
    self = [super init];
    if(!self) {
        return nil;
    }
    
    self.ID = ID;
    self.text = text;
    self.name = name;
    self.tintColor = tintColor;
    self.baseColor = baseColor;
    self.imageName = imageName;
    
    return self;
}

@end

File: Blaze-Swift/Objects/BlazeInputTile.swift
//
//  BlazeInputTile.swift
//  Blaze-Swift
//
//  Created by Sebastiaan Seegers on 10-11-17.
//  Copyright © 2017 ZoinksInteractive. All rights reserved.
//

import UIKit

class BlazeInputTile: NSObject {
    
    var ID: Int = 0
    var name: String?
    var text: String?
    var tintColor: UIColor?
    var baseColor: UIColor?
    var imageName: String?
    
    convenience init(withID ID: Int, Text text: String, TintColor tintColor: UIColor, BaseColor baseColor: UIColor, ImageName imageName: String) {
        self.init(withID: ID, Text: text, TintColor: tintColor, BaseColor: baseColor, ImageName: imageName)
    }
    
    convenience init(withID ID: Int, Name name: String, Text text: String, TintColor tintColor: UIColor, BaseColor baseColor: UIColor, ImageName imageName: String) {
        
        self.init()
        
        self.ID = ID
        self.text = text
        self.name = name
        self.tintColor = tintColor
        self.baseColor = baseColor
        self.imageName = imageName
    }
}
